include/pybind11/pybind11.h

```diff
@@ -143,12 +143,12 @@ class cpp_function : public function {
             /* Without these pragmas, GCC warns that there might not be
                enough space to use the placement new operator. However, the
                'if' statement above ensures that this is the case. */
-#if defined(__GNUG__) && !defined(__clang__) && __GNUC__ >= 6
+#if defined(__GNUG__) && !defined(__clang__) && __GNUC__ >= 6 && !defined(__INTEL_COMPILER)
```

Comment:
Very minor, but what do you think about making this like below, to not mix the "but not gcc compilers other than gcc" logic with specifics about gcc?
```
#if defined(__GNUG__) && __GNUC__ >= 6 && !defined(__clang__) && !defined(__INTEL_COMPILER)
```

Reply:
I agree this way it is nicer